js/util/string_number_mapping.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = StringNumberMapping;
+
+function StringNumberMapping(strings) {
+    this.stringToNumber = {};
+    this.numberToString = [];
+    for (var i = 0; i < strings.length; i++) {
+        var string = strings[i];
+        this.stringToNumber[string] = i;
+        this.numberToString[i] = string;
+    }
+}
```

Comment:
I see. My confusion stemmed from this module's name. Brainstorming some alternates now...